### ISSUE TITLE
Command: oryx build takes forever

### ISSUE THREAD

**Issue by martinhesdahl** [microsoft/Oryx#1390]
When deploying a python function to an azure dedicated function plan the yaml deployment task 'AzureFunctionApp@1' takes forever (We canceled it after 24 minutes) to complete.

This is the log details:

```
Command: oryx build /tmp/zipdeploy/extracted -o /tmp/build/expressbuild --platform python --platform-version 3.7 -i /tmp/8da38a5849dbfcf -p packagedir=.python_packages/lib/site-packages
Operation performed by Microsoft Oryx, https://github.com/Microsoft/Oryx
You can report issues at https://github.com/Microsoft/Oryx/issues

Oryx Version: 0.2.20211207.1, Commit: 46633df49cc8fbe9718772a3c894df221273b2af, ReleaseTagName: 20211207.1

Build Operation ID: |n5l2M7ajiVU=.8a681207_
Repository Commit : 63f7eccb-00b2-4086-a1c9-e31a3a43e4ea

Detecting platforms...
Detected following platforms:
  python: 3.7.12
Warning: An outdated version of python was detected (3.7.12). Consider updating.\nVersions supported by Oryx: https://github.com/microsoft/Oryx


Using intermediate directory '/tmp/8da38a5849dbfcf'.

Copying files to the intermediate directory...
Done in 58 sec(s).

Source directory     : /tmp/8da38a5849dbfcf
Destination directory: /tmp/build/expressbuild

Python Version: /opt/python/3.7.12/bin/python3.7
Creating directory for command manifest file if it doesnot exist
Removing existing manifest file

Running pip install...
[08:09:16+0000] Collecting azure-functions==1.6.0
[08:09:16+0000]   Downloading azure_functions-1.6.0-py3-none-any.whl (126 kB)
[08:09:16+0000] Collecting cycler==0.10.0
[08:09:16+0000]   Downloading cycler-0.10.0-py2.py3-none-any.whl (6.5 kB)
[08:09:17+0000] Collecting kiwisolver==1.3.1
[08:09:17+0000]   Downloading kiwisolver-1.3.1-cp37-cp37m-manylinux1_x86_64.whl (1.1 MB)
[08:09:18+0000] Collecting matplotlib==3.3.4
[08:09:18+0000]   Downloading matplotlib-3.3.4-cp37-cp37m-manylinux1_x86_64.whl (11.5 MB)
[08:09:25+0000] Collecting numpy==1.19.5
[08:09:25+0000]   Downloading numpy-1.19.5-cp37-cp37m-manylinux2010_x86_64.whl (14.8 MB)
[08:09:30+0000] Collecting pandas==1.1.5
[08:09:30+0000]   Downloading pandas-1.1.5-cp37-cp37m-manylinux1_x86_64.whl (9.5 MB)
[08:09:36+0000] Collecting Pillow==8.1.0
[08:09:36+0000]   Downloading Pillow-8.1.0-cp37-cp37m-manylinux1_x86_64.whl (2.2 MB)
[08:09:37+0000] Collecting pyodbc==4.0.30
[08:09:37+0000]   Downloading pyodbc-4.0.30.tar.gz (266 kB)
[08:09:39+0000] Collecting pyparsing==2.4.7
[08:09:39+0000]   Downloading pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
[08:09:39+0000] Collecting python-dateutil==2.8.1
[08:09:39+0000]   Downloading python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
[08:09:41+0000] Collecting pytz==2021.1
[08:09:42+0000]   Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)
[08:09:45+0000] Collecting six==1.15.0
[08:09:45+0000]   Downloading six-1.15.0-py2.py3-none-any.whl (10 kB)
[08:09:45+0000] Collecting Flask==2.0.2
[08:09:45+0000]   Downloading Flask-2.0.2-py3-none-any.whl (95 kB)
[08:09:46+0000] Collecting pydantic==1.8.2
[08:09:46+0000]   Downloading pydantic-1.8.2-cp37-cp37m-manylinux2014_x86_64.whl (10.1 MB)
[08:09:49+0000] Collecting azure-servicebus==7.3.2
[08:09:49+0000]   Downloading azure_servicebus-7.3.2-py2.py3-none-any.whl (200 kB)
[08:09:49+0000] Collecting arrow==1.2.1
[08:09:49+0000]   Downloading arrow-1.2.1-py3-none-any.whl (63 kB)
[08:09:50+0000] Collecting testfixtures==6.18.3
[08:09:50+0000]   Downloading testfixtures-6.18.3-py2.py3-none-any.whl (95 kB)
[08:09:52+0000] Collecting openpyxl==3.0.9
[08:09:52+0000]   Downloading openpyxl-3.0.9-py2.py3-none-any.whl (242 kB)
[08:09:54+0000] Collecting click>=7.1.2
[08:09:54+0000]   Downloading click-8.1.3-py3-none-any.whl (96 kB)
[08:09:54+0000] Collecting Jinja2>=3.0
[08:09:54+0000]   Downloading Jinja2-3.1.2-py3-none-any.whl (133 kB)
[08:09:56+0000] Collecting Werkzeug>=2.0
[08:09:56+0000]   Downloading Werkzeug-2.1.2-py3-none-any.whl (224 kB)
[08:09:57+0000] Collecting itsdangerous>=2.0
[08:09:57+0000]   Downloading itsdangerous-2.1.2-py3-none-any.whl (15 kB)
[08:09:57+0000] Collecting typing-extensions>=3.7.4.3
[08:09:58+0000]   Downloading typing_extensions-4.2.0-py3-none-any.whl (24 kB)
[08:09:59+0000] Collecting azure-core<2.0.0,>=1.14.0
[08:09:59+0000]   Downloading azure_core-1.24.0-py3-none-any.whl (178 kB)
[08:09:59+0000] Collecting msrest<2.0.0,>=0.6.17
[08:09:59+0000]   Downloading msrest-0.6.21-py2.py3-none-any.whl (85 kB)
[08:10:11+0000] Collecting uamqp<2.0.0,>=1.4.1
[08:10:11+0000]   Downloading uamqp-1.5.3-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (3.1 MB)
[08:10:15+0000] Collecting isodate>=0.6.0
[08:10:15+0000]   Downloading isodate-0.6.1-py2.py3-none-any.whl (41 kB)
[08:10:17+0000] Collecting azure-common~=1.1
[08:10:17+0000]   Downloading azure_common-1.1.28-py2.py3-none-any.whl (14 kB)
[08:10:18+0000] Collecting et-xmlfile
[08:10:18+0000]   Downloading et_xmlfile-1.1.0-py3-none-any.whl (4.7 kB)
[08:10:23+0000] Collecting importlib-metadata; python_version < "3.8"
[08:10:23+0000]   Downloading importlib_metadata-4.11.3-py3-none-any.whl (18 kB)
[08:10:27+0000] Collecting MarkupSafe>=2.0
[08:10:27+0000]   Downloading MarkupSafe-2.1.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
[08:10:29+0000] Collecting requests>=2.18.4
[08:10:29+0000]   Downloading requests-2.27.1-py2.py3-none-any.whl (63 kB)
[08:10:30+0000] Collecting certifi>=2017.4.17
[08:10:30+0000]   Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
[08:10:32+0000] Collecting requests-oauthlib>=0.5.0
[08:10:32+0000]   Downloading requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
[08:10:33+0000] Collecting zipp>=0.5
[08:10:33+0000]   Downloading zipp-3.8.0-py3-none-any.whl (5.4 kB)
[08:10:38+0000] Collecting urllib3<1.27,>=1.21.1
[08:10:38+0000]   Downloading urllib3-1.26.9-py2.py3-none-any.whl (138 kB)
[08:10:40+0000] Collecting charset-normalizer~=2.0.0; python_version >= "3"
[08:10:40+0000]   Downloading charset_normalizer-2.0.12-py3-none-any.whl (39 kB)
[08:10:42+0000] Collecting idna<4,>=2.5; python_version >= "3"
[08:10:42+0000]   Downloading idna-3.3-py3-none-any.whl (61 kB)
[08:10:43+0000] Collecting oauthlib>=3.0.0
[08:10:43+0000]   Downloading oauthlib-3.2.0-py3-none-any.whl (151 kB)
[08:10:43+0000] Building wheels for collected packages: pyodbc
[08:10:44+0000]   Building wheel for pyodbc (setup.py): started
```

Could you guys take a look at why this process keeps on running?

**Comment by cormacpayne** [comment]
@martinhesdahl Hey there, sorry that you're running into these long build times -- are you able to provide a link to the repository that's hosting the above Python Azure Function application, or if not, maybe provide some information about the application itself and we can attempt to reproduce this issue locally?

Also, is this a one-off issue that you saw with the `AzureFunctionApp@1` task, or do the builds consistently take a long time from multiple runs of this task against your Python Azure Function application?

**Comment by martinhesdahl** [comment]
Hello @cormacpayne,

Thanks for getting back to me. It is consistent behaviour that we saw with the AzureFunctionApp@1 task. I have to mention though that the behaviour was only displayed for dedicated function apps. 
When I changed back from dedicated plan to consumption plan the long build times did not occur. For consumption plan it was around ~30 seconds to finish the AzureFunctionApp@1 task.

I'm terribly sorry but I can't provide you with link to our repository. I can however, tell you that the repo contains four python functions of which two timer triggers, one queue triggered and one http triggered. 

They used to be consumption plan and when I changed to a dedicated plan through an ARM-template the long build times occurred consistently. I can share you a clean arm template and the yaml I used to deploy in devops.
Of course I deleted the resources first from the Azure portal before deploying the new arm template with the changed service plan.

Tomorrow is my day off so I'll see if I can set up an open repository for you to reproduce the matter.
